llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -13644,24 +13644,17 @@ struct NodeExtensionHelper {
     SupportsFPExt = false;
     EnforceOneUse = true;
     CheckMask = true;
+    SDValue NarrowOp;
     unsigned Opc = OrigOperand.getOpcode();
     switch (Opc) {
     case ISD::ZERO_EXTEND:
     case ISD::SIGN_EXTEND: {
+      NarrowOp = OrigOperand.getOperand(0);
```

Comment:
Can't se just check for i1 in this case? None of the other cases can have an i1 narrow op past type legalization.